src/lando/utils/github_checks.py

```diff
@@ -193,6 +193,66 @@ def run(
 #     def run(cls, client: GitHubAPIClient, pull_request: PullRequest, target_repo: Repo, request: HttpRequest, request: HttpRequest) -> list[str]:
 #         raise NotImplementedError
 
+# GITHUB-SPECIFIC CHECKS
+
+
+class PullRequestBaseBranchDoesntMatchTree(PullRequestBlocker):
+    """The base branch for this PR doesn't match this Tree."""
+
+    @override
+    @classmethod
+    def run(
+        cls,
+        client: GitHubAPIClient,
+        pull_request: PullRequest,
```

Comment:
nit: this PR should probably extend #644 so that it can utilize `pull_request._client`. We should also remove the leading underscore since the client may be used outside of the pull request object.